packages/foam-vscode/global-setup.js

```diff
@@ -0,0 +1,5 @@
+const setTZ = require('set-tz')
+
+module.exports = async () => {
+    setTZ('UTC')
```

Comment:
what are you trying to achieve here and what problem is it solving?

Reply:
Without this code, if the tests are run locally on a developer's machine, and their computer's timezone is not 'UTC', the tests may not pass.

This is because `dateFormat` will use the date of the local timezone when formatting the filename that is used for the daily note's filepath.

For example, if the time is `2021-02-07T00:00:00Z`, it is only `2021-02-06T19:00:00-05:00` in `Americas/Toronto`, so if a developer in `Americas/Toronto` runs the tests locally, the test will fail. The test expects the filename to be `2021-02-07.md`, but the developer's local timezone will cause `dateFormat` to produce a filename of `2021-02-06.md`.

In order to have tests that succeed regardless of the developer's timezone, we need to fix the timezone used by the tests. `set-tz` does this in a way that is cross-platform (since fixing the timezone is not the same in Windows (much harder) as it is on Mac or Linux (much easier)).

Reply:
thanks for the explanation, sometimes I wonder whether we should use moment.js, but so far it's not been super necessary.

Back to my original point, the reason why I was asking was because we don't use TZ in our code.
So, I feel that we are now introducing TZ in the test code, which then we need to account for.

Let me elaborate:
what if instead of using `const date = new Date('2021-02-07T00:00:00Z');`, you used `const date = new Date();`, then generate the expected folder path? (you can also be specific by assigning day/month/year if you want to make the test always about the same value)

This way I think you wouldn't need to deal with TZ issues in the first place.

I am not against adding dependencies per se, but I lean towards avoiding it if possible to keep the code easier to reason about without knowing its ins and outs - "if possible" being the key words there :)

wdyt? open to hearing thoughts both on the specific case and on the general approach

Reply:
Yeah, makes sense.

My only hesitation is a general reluctance to put non-determinism in test suites.

But in this case, perhaps doesn't justify the new dependency (a reason why JS/TS development is weird... everything is a dependency).

I'll look into it later this week.

Reply:
Alright, I now manually create the corresponding filename in the the tests instead of fixing the timezone. So I've removed the `set-tz` dependency.